packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -540,6 +678,13 @@ class Baseline extends OneChildRenderObjectWidget {
   }
 }
 
+/// A widget that's bigger on the inside
+///
+/// A viewport is the core scrolling primitive in the render tree. The child of
+/// a viewport can layout to a larger size than the viewport itself. If that
+/// happens, only a portion of the child will be visible through the viewport.
+/// The portion of the child that is visible is controlled by the scroll
+/// offset.
```

Comment:
we should add something to mention that you can also use this in non-scrolling contexts, because I didn't really realise that for a long time.